Add options for timing and charge resolutions for PMT hits

tdealtry opened the pull request:
Implemented by truncating the floating point values to the user-specified precision
See #188 for discussion

Standard verification. KS tests are 1

![electrontest_digit_time_rounding](https://cloud.githubusercontent.com/assets/11757479/19472397/09f96258-951f-11e6-9639-9a58823c6cbf.png)

eosullivan commented:
There should be a small change in the timing plots, since there is currently no rounding and you are now introducing digitizer-specific rounding, right? Is it just not showing up in your validation plots because the change is small? Can you confirm when you make a large change (i.e. a large timing resolution) that the code behaves as you expect? And maybe also test the charge smearing as well, even though it's default is not being changed here?

tdealtry commented:
You were right. Fixed it now. I've also added the new options to `WCSimRootOptions` So it's ready to go.


Running with:
```
/DAQ/DigitizerOpt/TimingPrecision 0.0
/DAQ/DigitizerOpt/PEPrecision 0.0
```
Gives
![electrontest_digit_time_rounding_off](https://cloud.githubusercontent.com/assets/11757479/26404235/08875374-4089-11e7-9abd-65954c0b136f.png)
```
ks test for # of digitized hits: 1
ks test for average charge: 1
ks test for charge: 1
ks test for average time: 1
ks test for time: 0
***********************************************************
mean for # of digitized hits  : 2091.22
mean for # of digitized hits 2: 2091.2
mean for average charge  : 2.02629
mean for average charge 2: 2.0263
mean for charge  : 1.90349
mean for charge 2: 1.9035
mean for average time  : 1000.47
mean for average time 2: 1000.48
mean for time  : 998.927
mean for time 2: 998.938
```
The difference in digit time is due to the redefinition of trigger time: the trigger time is no longer rounded to an `int`, but instead store as `float`. Therefore `digit_time = hit_time - trigger_time` is smoother due to the lack of discontinues in `trigger_time`

Running with:
```
/DAQ/DigitizerOpt/TimingPrecision 0.4
/DAQ/DigitizerOpt/PEPrecision 0.1
```
Gives
![electrontest_digit_time_rounding_on](https://cloud.githubusercontent.com/assets/11757479/26397583/b1c473ce-406e-11e7-875d-a4c997b1d5d3.png)
```
ks test for # of digitized hits: 1
ks test for average charge: 2.545e-33
ks test for charge: 0
ks test for average time: 0.981434
ks test for time: 0
***********************************************************
mean for # of digitized hits  : 2091.22
mean for # of digitized hits 2: 2091.21
mean for average charge  : 2.02629
mean for average charge 2: 1.9763
mean for charge  : 1.90349
mean for charge 2: 1.8535
mean for average time  : 1000.47
mean for average time 2: 1000.31
mean for time  : 998.927
mean for time 2: 998.772
```
You can clearly see the effect of truncating the charge

eosullivan commented:
Ah, I forgot that this PR was ready to go. I'll merge this in so we can have this implemented for the production runs.
